src/verification_method.rs

```diff
@@ -29,19 +37,72 @@ pub struct VerificationMethod {
     pub verification_relationships: Vec<VerificationRelationship>,
 }
 
-pub fn format_verification_method(
+pub fn add_verification_methods(
     e: &Env,
     verification_methods: &Vec<VerificationMethod>,
     did_uri: &String,
-) -> Vec<VerificationMethod> {
-    let mut new_verification_methods: Vec<VerificationMethod> = Vec::new(e);
+    did_document: &mut DIDDocument,
+) {
+    did_document.verification_method = format_verification_method(e, verification_methods, did_uri);
+    clean_all_relationships(e, did_document);
+
+    for verification_method in verification_methods.iter() {
+        for relationship in verification_method.verification_relationships {
+            let value = vec![
+                e,
+                did_uri::concat_fragment(e, did_uri, &verification_method.id),
+            ];
+            format_relationship(did_document, relationship, value);
+        }
+    }
+}
```

Comment:
Is this function able to add multiple verification methods per relationship?
For example, having two keys for "authentication".

It seems like it will always have only 1...

Reply:
Yes, it iterates through all the verification methods in the vector and then iterates by relationship.